cloud/amazon/ec2_eni.py

```diff
@@ -97,6 +108,29 @@
     subnet_id: subnet-xxxxxxxx
     state: present
 
+# Create an ENI with two secondary addresses
+- ec2_eni:
+    subnet_id: subnet-xxxxxxxx
+    state: present
+    secondary_private_ip_address_count: 2
+
+# Assign a secondary IP address to an existing ENI
+# This will purge any existing IPs
+- ec2_eni:
+    subnet_id: subnet-xxxxxxxx
+    eni_id: eni-yyyyyyyy
+    state: present
+    secondary_private_ip_addresses:
+      - 172.16.1.1
+
+# Remove any secondary IP addresses from an existing ENI
+- ec2_eni:
+    subnet_id: subnet-xxxxxxxx
+    eni_id: eni-yyyyyyyy
+    state: present
+    secondary_private_ip_addresses:
+      - 
```

Comment:
Nope.  Single empty item in a list will remove any secondary IP. This is
mentioned in the doc.

The default is an empty list and I don't want to remove secondary IPs that
may of been defined just because someone didn't declare the parameter.
On 6 May 2016 4:24 AM, "René Moser" <notifications@github.com> wrote:

> In cloud/amazon/ec2_eni.py
> <https://github.com/ansible/ansible-modules-extras/pull/2161#discussion_r62230894>
> :
>
> > +# Assign a secondary IP address to an existing ENI
> > +# This will purge any existing IPs
> > +- ec2_eni:
> > +    subnet_id: subnet-xxxxxxxx
> > +    eni_id: eni-yyyyyyyy
> > +    state: present
> > +    secondary_private_ip_addresses:
> > +      - 172.16.1.1
> > +
> > +# Remove any secondary IP addresses from an existing ENI
> > +- ec2_eni:
> > +    subnet_id: subnet-xxxxxxxx
> > +    eni_id: eni-yyyyyyyy
> > +    state: present
> > +    secondary_private_ip_addresses:
> > +      -
>
> this looks like there is something missing
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible-modules-extras/pull/2161/files/f56a0ce1c51040c42feeed654d739858078519e8#r62230894>
>